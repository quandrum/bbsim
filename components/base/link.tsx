import { AnchorHTMLAttributes } from 'react';

// TODO turn in button for a11y
export const Link: React.FC<AnchorHTMLAttributes<HTMLAnchorElement>> = ({
  onClick,
  children,
}) => (
  <a
    href="#"
    className="mb-6 mr-auto text-base leading-normal text-white font-roboto hover:underline"
    onClick={onClick}
  >
    {children}
  </a>
);
