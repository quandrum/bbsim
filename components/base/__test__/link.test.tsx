import { render, screen, fireEvent } from '@testing-library/react';
import { Link } from '../link';

describe('Link', () => {
  const DEFAULT_PROPS = {
    onClick: jest.fn(),
  };

  beforeEach(() => {
    DEFAULT_PROPS.onClick.mockReset();
  });

  it('renders', () => {
    const { container } = render(<Link {...DEFAULT_PROPS}>test</Link>);
    expect(container.innerHTML).toMatchSnapshot();
  });
  it('calls onClick when clicked', () => {
    const text = 'test';
    render(<Link {...DEFAULT_PROPS}>{text}</Link>);
    fireEvent.click(screen.getByText(text));
    expect(DEFAULT_PROPS.onClick).toHaveBeenCalled();
  });
});
