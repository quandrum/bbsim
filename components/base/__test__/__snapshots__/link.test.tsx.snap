// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders 1`] = `"<a href=\\"#\\" class=\\"mb-6 mr-auto text-base leading-normal text-white font-roboto hover:underline\\">test</a>"`;
